src/buffer/out/textBuffer.cpp

```diff
@@ -418,12 +479,20 @@ bool TextBuffer::InsertCharacter(const std::wstring_view chars,
         auto& Row = GetRowByOffset(iRow);
 
         // Store character and double byte data
-        auto& charRow = Row.GetCharRow();
-
         try
         {
-            charRow.GlyphAt(iCol) = chars;
-            charRow.DbcsAttrAt(iCol) = dbcsAttribute;
+            switch (dbcsAttribute)
+            {
+            case DbcsAttribute::Leading:
+                Row.ReplaceCharacters(iCol, 2, chars);
+                break;
+            case DbcsAttribute::Trailing:
+                Row.ReplaceCharacters(iCol - 1, 2, chars);

```

Comment:
There's a coded assumption here that we will almost certainly miss, which is that a trailing column is always part of a _two cell wide_ glyph. This will break the day we add widths greater than two.

Reply:
Good point. Despite that I feel like this code is correct given the current DBCS assumptions throughout the project, all of which sorta assume that only a single trailing one can follow a single leading one.